pkg/vendir/fetch/hg/hg.go

```diff
@@ -134,6 +194,7 @@ hgauth.password = %s
 			}
 
 			sshCmd = append(sshCmd, "-o", "StrictHostKeyChecking=yes", "-o", "UserKnownHostsFile="+path)
+			cacheID += "known-hosts=" + *authOpts.KnownHosts + "|"
```

Comment:
What are we trying to prevent with the KownHosts inclusion in the CacheID?

Reply:
Nothing really, I realize the only key we need is the repository URL. Will do that.